src/topology/metric_space/emetric_space.lean

```diff
@@ -307,7 +307,16 @@ instance emetric_space_pi [∀b, emetric_space (π b)] : emetric_space (Πb, π
       have eq1 : sup univ (λ (b : β), edist (f b) (g b)) ≤ 0 := le_of_eq eq0,
       simp only [finset.sup_le_iff] at eq1,
       exact (funext $ assume b, eq_of_edist_eq_zero $ bot_unique $ eq1 b $ mem_univ b),
-    end }
+    end,
+  to_uniform_space := Pi.uniform_space _,
+  uniformity_edist := begin
+    -- with simp only on next line, the proof fails for no reason...
```

Comment:
You can use `simp only` here, and add `change 0 < ε at εpos,` before the last line.

Reply:
Done